tests/AstrolabeWorkloadExecutor/AstrolabeTestRunner.cs

```diff
@@ -0,0 +1,204 @@
+/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers;
+using MongoDB.Bson.TestHelpers.JsonDrivenTests;
+using MongoDB.Driver;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.TestHelpers;
+using MongoDB.Driver.Tests.Specifications.Runner;
+
+namespace WorkloadExecutor
+{
+    public class AstrolabeTestRunner : MongoClientJsonDrivenTestRunnerBase
+    {
+        // private fields
+        private readonly CancellationToken _cancellationToken;
+        private readonly Action _incrementOperationSuccesses;
+        private readonly Action _incrementOperationErrors;
+        private readonly Action _incrementOperationFailures;
+
+        // protected properties
+        protected override string[] ExpectedSharedColumns => new[] { "_path", "database", "collection", "testData", "tests" };
+        protected override string[] ExpectedTestColumns => new[] { "operations", "async" };
+
+        public AstrolabeTestRunner(
+            Action incrementOperationSuccesses,
+            Action incrementOperationErrors,
+            Action incrementOperationFailures,
+            CancellationToken cancellationToken)
+        {
+            _incrementOperationSuccesses = incrementOperationSuccesses;
+            _incrementOperationErrors = incrementOperationErrors;
+            _incrementOperationFailures = incrementOperationFailures;
+            _cancellationToken = cancellationToken;
+        }
+
+        protected override string DatabaseNameKey => "database";
+        protected override string CollectionNameKey => "collection";
+        protected override string DataKey => "testData";
+
+        // public methods
+        public void Run(JsonDrivenTestCase testCase)
+        {
+            SetupAndRunTest(testCase);
+        }
+
+        // protected methods
+        protected override void AssertOperation(JsonDrivenTest test)
+        {
+            var wrappedActualException = test._actualException();
```

Comment:
Why is this variable called `wrappedActualException`?

Seems like the correct name should be just `actualException`.

There is no wrapping going on.